File: solve/golang/10828/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	scanner.Scan()
	n, err := strconv.Atoi(scanner.Text())
	if err != nil {
		panic(err)
	}

	stack := []int{}

	for i := 0; i < n; i++ {
		scanner.Scan()
		cmd := strings.Fields(scanner.Text())

		switch cmd[0] {
		case "push":
			val, err := strconv.Atoi(cmd[1])
			if err != nil {
				panic(err)
			}
			stack = append(stack, val)
		case "pop":
			if len(stack) == 0 {
				fmt.Fprintln(writer, -1)
			} else {
				fmt.Fprintln(writer, stack[len(stack)-1])
				stack = stack[:len(stack)-1]
			}
		case "size":
			fmt.Fprintln(writer, len(stack))
		case "empty":
			if len(stack) == 0 {
				fmt.Fprintln(writer, 1)

			} else {
				fmt.Fprintln(writer, 0)
			}
		case "top":
			if len(stack) == 0 {
				fmt.Fprintln(writer, -1)
			} else {
				fmt.Fprintln(writer, stack[len(stack)-1])
			}
		}
	}
}

File: solve/CPP/10828/main.cpp

#include <iostream>
#include <string>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;

    int b[10000];
    int bcount = 0;

    string cmd;
    for (int i = 0; i < N; i++) {
        cin >> cmd;

        if (cmd == "push") {
            int x;
            cin >> x;
            b[bcount] = x;
            bcount++;
        }
        else if (cmd == "pop") {
            if (bcount == 0) {
                cout << -1 << '\n';
            } else {
                cout << b[bcount - 1] << '\n';
                bcount--;
            }
        }
        else if (cmd == "size") {
            cout << bcount << '\n';
        }
        else if (cmd == "empty") {
            cout << (bcount == 0 ? 1 : 0) << '\n';
        }
        else if (cmd == "top") {
            if (bcount == 0) cout << -1 << '\n';
            else cout << b[bcount - 1] << '\n';
        }
    }

    return 0;
}
